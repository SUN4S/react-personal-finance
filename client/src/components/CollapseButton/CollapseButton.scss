nav {
  .collapseButton {
    height: 46px;
    width: 46px;
    background-color: var(--sidebar-bg);
    border: none;
    align-self: flex-end;
    color: var(--text-primary);
    font-size: 18px;
    transition: all 0.5s ease;

    &:hover {
      cursor: pointer;
    }
  }
  &.collapsed {
    .collapseButton {
      transform: rotateZ(180deg);
    }
  }
}
