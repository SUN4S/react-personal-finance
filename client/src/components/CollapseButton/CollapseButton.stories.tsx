import { Meta, Story } from "@storybook/react/types-6-0";

import { CollapseButton } from "./CollapseButton";

interface TemplateProps {
  theme: string;
  onClick: Function;
}

export default {
  title: "Components/Collapse Button",
  component: CollapseButton,
} as Meta;

const Template: Story<TemplateProps> = (args) => {
  return (
    <html data-theme={args.theme}>
      <body style={{ height: "50px", width: "50px" }}>
        <CollapseButton clickFunction={args.onClick} />
      </body>
    </html>
  );
};

export const Default_Light = Template.bind({});
Default_Light.args = {
  theme: "light",
  onClick: () => alert("clicked"),
};

export const Default_Dark = Template.bind({});
Default_Dark.args = {
  theme: "dark",
  onClick: () => alert("clicked"),
};
