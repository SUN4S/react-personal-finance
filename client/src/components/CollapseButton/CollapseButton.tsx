import "./CollapseButton.scss";

import { IconArrow } from "../Icons/IconArrow/IconArrow";

export const CollapseButton = (props: { clickFunction: Function }) => {
  return (
    <button
      className="collapseButton"
      data-testid="collapseButton"
      onClick={() => props.clickFunction()}
    >
      <IconArrow />
    </button>
  );
};
