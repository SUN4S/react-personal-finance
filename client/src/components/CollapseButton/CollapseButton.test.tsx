import { cleanup, fireEvent, render } from "@testing-library/react";

import { CollapseButton } from "./CollapseButton";
import renderer from "react-test-renderer";

describe("Collapse Button", () => {
  afterEach(cleanup);

  it("Renders", () => {
    const mockFunction = jest.fn();
    const { getByTestId } = render(
      <CollapseButton clickFunction={mockFunction} />
    );
    expect(getByTestId("collapseButton")).toBeTruthy();
  });

  it("Fires Function", () => {
    const mockFunction = jest.fn();
    const { getByTestId } = render(
      <CollapseButton clickFunction={mockFunction} />
    );
    fireEvent.click(getByTestId("collapseButton"));
    expect(mockFunction).toHaveBeenCalledTimes(1);
  });

  it("Matches Snapshot", () => {
    const mockFunction = jest.fn();
    const tree = renderer
      .create(<CollapseButton clickFunction={mockFunction} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
