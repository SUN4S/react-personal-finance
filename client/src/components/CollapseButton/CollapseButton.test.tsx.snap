// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse Button Matches Snapshot 1`] = `
<button
  className="collapseButton"
  data-testid="collapseButton"
  onClick={[Function]}
>
  <svg
    id="arrowIcon"
    viewBox="0 0 52 36"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      data-name="Layer 2"
      id="Layer_2"
    >
      <g
        data-name="Layer 1"
        id="Layer_1-2"
      >
        <line
          className="cls-1"
          x1="1"
          x2="51"
          y1="17.92"
          y2="17.92"
        />
        <line
          className="cls-1"
          x1="51"
          x2="26"
          y1="18"
          y2="1"
        />
        <line
          className="cls-1"
          x1="51"
          x2="26"
          y1="17.83"
          y2="34.83"
        />
      </g>
    </g>
  </svg>
</button>
`;
